src/consumer/mod.rs

```diff
@@ -243,7 +263,7 @@ impl<E> ConsumerBuilder<E> {
         K: Into<String>,
         H: Fn(Job) -> Result<(), E> + Send + Sync + 'static,
     {
-        self.register_runner(kind, handler)
+        self.register_runner(kind, Box::new(handler))
```

Comment:
I think we can avoid the boxing here for closures by adding a private type

```rust
#[repr(transparent)]
struct Closure<H>(H);
impl<E, H> JobRunner for Closure<H>
    where H: Fn(Job) -> Result<(), E> + Send + Sync + 'static
{
    type Error = E;
    fn run(&self, job: Job) -> Result<(), Self::Error> {
        (self.0)(job)
    }
}
```

and then passing `Closure(handler)` here